### ISSUE TITLE
Undocumented ncurses dependency in supdup

### ISSUE THREAD

**Issue by mseddon** [PDP-10/its#1813]
After several hours of building on my vm, I ran into the following:
```
Sat 01 Feb 2020 10:31:31 AM PST
touch out/pdp10-kl/stamp
make -C tools/supdup
make[1]: Entering directory '/home/matt/its/its/tools/supdup'
cc -g -Wall   -c -o supdup.o supdup.c
supdup.c:58:10: fatal error: curses.h: No such file or directory
   58 | #include <curses.h>
      |          ^~~~~~~~~~
compilation terminated.
make[1]: *** [<builtin>: supdup.o] Error 1
make[1]: Leaving directory '/home/matt/its/its/tools/supdup'
make: *** [Makefile:263: tools/supdup/supdup] Error 2
```

After installing ncurses, a re-run of make appears to be starting over, though I am not certain:

```
make EMULATOR=pdp10-kl
git submodule sync `dirname tools/tv11/.gitignore`
Synchronizing submodule url for 'tools/tv11'
git submodule update --init `dirname tools/tv11/.gitignore`
git submodule sync `dirname tools/pdp6/.gitignore`
Synchronizing submodule url for 'tools/pdp6'
git submodule update --init `dirname tools/pdp6/.gitignore`
mkdir -p out/pdp10-kl/_klfe_
tools/dasm/klfedr > "out/pdp10-kl/_klfe_/kldcp.(dir)"
mkdir -p out/pdp10-kl
tools/itstar/itstar -cf out/pdp10-kl/kl-minsys.tape -C out/pdp10-kl _klfe_
tools/itstar/itstar -rf out/pdp10-kl/kl-minsys.tape -C bin/kl10 _ sys
tools/itstar/itstar -rf out/pdp10-kl/kl-minsys.tape -C bin/minsys sys
expect -f build/pdp10-kl/build.tcl 192.168.1.100 192.168.0.45

ENTERING MAIN BUILD SCRIPT
Sat 01 Feb 2020 10:56:21 AM PST


ENTERING BUILD SCRIPT: MARK
Sat 01 Feb 2020 10:56:21 AM PST

spawn ./tools/sims/BIN/pdp10-kl build/pdp10-kl/init

KL-10 simulator V4.0-0 Current        git commit id: 88b26df8
/home/matt/its/its/build/pdp10-kl/init-15> at rpa1 out/pdp10-kl/rp04.1
RPA: creating new file
MAGDMP
l$ddt
t$salv bin
MARK$G

FORMAT PACK ON UNIT #0
ARE YOU SURE YOU WANT TO FORMAT PACK ON DRIVE #0? Y
PACK NO =0
FORMATTING COMPLETE, VERIFICATION BEGINS

SWAPPING ALLOC =3000
PACK # =0
PACK ID =0
DDT
```

Either way the rp06(?)'s are being reformatted.

This is not a huge deal for me, but a mention of the ncurses dependency would help, (This is my first attempt at a build, so I'm not familiar enough to see if ALL or just some of the system is now being rebuilt from scratch again).

**Comment by mseddon** [comment]
Update: fwiw, I seem to be at the final tape creation stage again after a few minutes:
```
:dump
DUMP  .446

_load links crdir sorry
FILE=*;* *

TAPE NO      1 CREATION DATE  200201
```
So it seems I have survived a lot of the MIDAS assembly stage again at least.

**Comment by larsbrinkhoff** [comment]
No, you are right.  It's rebuilding.

By way of explanation: The build scripts, including the Makefile, are mostly geared to build everything.  Incremental builds are not really supported.  Of course it's a bug that a failed supdup should case a new make invokation to rebuild everything again.

**Comment by mseddon** [comment]
No problem at all- I can now report that the fairly minimal vm ubuntu install I have, only set up to run Symbolics Genera, (I had to even install git to get started with this!), has managed to succeed building, so if there are any other undocumented deps, they haven't affected me.

**Comment by mseddon** [comment]
I may be naive in suggesting- if there were seperate makefiles for the emulator binaries and the ITS tape/install process for those champing at the bit to get started, but honestly I would hate to lose that expect build system. Can it be pre-installed on a set of virtual disk-packs and distributed that way potentially?

**Comment by mseddon** [comment]
As an update, can confirm the system is perfectly usable. Having trouble with the sdl terminal emulators atm but is probably just PEBKAB on my part.

**Comment by larsbrinkhoff** [comment]
The combination xterm + TCTYP AAA works fine.  But not all terminal
emulators will handle everything ITS thinks AAA can do.

**Comment by larsbrinkhoff** [comment]
Fixed in #2017.
